app/assets/javascripts/analytics.js

```diff
@@ -1,60 +1,68 @@
-        if (!userId) {
-            userId = generateEightHexadecimalDigits() + generateEightHexadecimalDigits();
-            GOVUK.setCookie('PIWIK_USER_ID', userId);
-        }
-        return userId;
+  function getPiwikVisitorIdCookie() {
+    var userId = GOVUK.getCookie('PIWIK_USER_ID');
+    if (!userId) {
+      userId = generateEightHexadecimalDigits() + generateEightHexadecimalDigits();
+      GOVUK.setCookie('PIWIK_USER_ID', userId);
     }
+    return userId;
+  }
 
-    var trackerUrl = $('#piwik-url').text();
-
-    if(!trackerUrl) {
-        return;
-    }
+  var trackerUrl = $('#piwik-url').text();
 
+  if (trackerUrl) {
     var siteId = $('#piwik-site-id').text();
     var customUrl = $('#piwik-custom-url').text();
     var enTitle = $('meta[name="verify|title"]').attr("content");
     var newVisit = $('#piwik-new-visit').length ? 1 : 0;
 
     var piwikAnalyticsQueue = [
-        ['appendToTrackingUrl', 'new_visit=' + newVisit],
-        ['setUserId', getPiwikVisitorIdCookie()],
-        ['setDocumentTitle', enTitle ],
-        ['trackPageView'],
-        ['enableLinkTracking'],
-        ['setTrackerUrl', trackerUrl],
-        ['setSiteId', siteId]
+      ['appendToTrackingUrl', 'new_visit=' + newVisit],
+      ['setUserId', getPiwikVisitorIdCookie()],
+      ['setDocumentTitle', enTitle],
+      ['trackPageView'],
+      ['enableLinkTracking'],
+      ['setTrackerUrl', trackerUrl],
+      ['setSiteId', siteId]
     ];
 
     var customVariablesString = $('#piwik-custom-variables').text();
     if (customVariablesString) {
-        var customVariables = JSON.parse(customVariablesString);
-        $.each(customVariables, function(index) {
-            piwikAnalyticsQueue.unshift([
-                'setCustomVariable',
-                customVariables[index].index,
-                customVariables[index].name,
-                customVariables[index].value,
-                customVariables[index].scope
-            ])
-        });
+      var customVariables = JSON.parse(customVariablesString);
+      $.each(customVariables, function (index) {
+        piwikAnalyticsQueue.unshift([
+          'setCustomVariable',
+          customVariables[index].index,
+          customVariables[index].name,
+          customVariables[index].value,
+          customVariables[index].scope
+        ])
+      });
     }
 
     if (customUrl) {
-        // customUrl needs to go at the beginning of the piwik array
-        piwikAnalyticsQueue.unshift(['setCustomUrl', customUrl]);
+      // customUrl needs to go at the beginning of the piwik array
+      piwikAnalyticsQueue.unshift(['setCustomUrl', customUrl]);
     }
 
     global._paq = piwikAnalyticsQueue;
+  }
+
+  var cross_gov_ga_tracker_id = $("#cross-gov-ga-tracker-id").text();
+  if (cross_gov_ga_tracker_id) {
+    var domain_list = JSON.parse($("#cross-gov-ga-domain-list").text());
+    ga("create", cross_gov_ga_tracker_id, "auto", "govuk_shared", {"allowLinker": true});
```

Comment:
I'm not sure if switching from `ga` to `window.ga` would be enough to satisfy Codacy? Otherwise, we could try to mimic the snippet - it should never get called since the snippet will have been run by now.

```suggestion
    if (!window['ga']) {
        window.ga = function() {
            if (!window.ga.q) {
                window.ga.q = [];
            }
            window.ga.q.push(arguments)
        }
    }
    ga("create", cross_gov_ga_tracker_id, "auto", "govuk_shared", {"allowLinker": true});
```

🤷‍♂

Reply:
Codacy is happy, will leave as it is for now :-)